polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -1077,20 +1084,20 @@ where
 			&state.active_leaves,
 		)
 		.map_err(AdvertisementError::Invalid)?;
+
 	if !per_relay_parent.collations.is_seconded_limit_reached(relay_parent_mode) {
 		return Err(AdvertisementError::SecondedLimitReached)
 	}
 
 	if let Some((candidate_hash, parent_head_data_hash)) = prospective_candidate {
-		let is_seconding_allowed = !relay_parent_mode.is_enabled() ||
```

Comment:
`!relay_parent_mode.is_enabled() ||` is there in case two nodes communicate with V2 but async backing is still off, meaning we don't need to check for `can_second` -- there's only one slot for candidate per relay parent anyway.

Also, seems like this branch won't be executed at all even with async backing **enabled** if collator chooses to communicate with v1, which opens some room for spamming the node with validation work.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/1920#discussion_r1362441842

> Also, seems like this branch won't be executed at all even with async backing **enabled** if collator chooses to communicate with v1, which opens some room for spamming the node with validation work.

You mean because we would add the collation without further checks?

Reply:
> https://github.com/paritytech/polkadot-sdk/pull/1920#discussion_r1362441842

In the previous version of this PR `use_non_prospective_chains_mode` was defined based on what collator told us, in the original code it relies on the runtime request from validators.

> You mean because we would add the collation without further checks?

The idea is to only request collation once the parent parachain block becomes backable. Until then, we don't do any unnecessary work. With this code collator can
1. Choose to use v1 networking
2. Send maximum number of sequentially based collations and validators will start requesting & validating without waiting for the "backable parent" part.

Reply:
yes, in the v1 networking version it should only accept `parent_head_data_hash == current_head_hash`.

Reply:
> yes, in the v1 networking version it should only accept `parent_head_data_hash == current_head_hash`.

Should be done now.

Reply:
Done @slumber